src/components/cc-orga-member-card/cc-orga-member-card.js

```diff
@@ -0,0 +1,531 @@
+ *
+ * @cssdisplay block
+ *
+ * @event {CustomEvent<ToggleEditing>} cc-orga-member-card:toggle-editing - Fires the `id` of the member related to the card and the new state (`editing` or `loaded`) to specify when card is in edit mode or not.
+ * This allows the list component to close all other cards in edit mode to leave only one in edit mode at once.
+ * @event {CustomEvent<DeleteMember>} cc-orga-member-card:delete - Fires the `id` of the member to be removed from the org with their `name` or `email`.
+ * @event {CustomEvent<DeleteMember>} cc-orga-member-card:leave - Fires the `id` of the member to be removed from the org with their `name` or `email`.
+ * We don't fire a delete event so that it can be processed differently by the smart component (leaving the org means the user has to be redirected).
+ * @event {CustomEvent<UpdateMember>} cc-orga-member-card:update - Fires the `id` of the member, their new `role` and their `name` or `email`.
+ */
+export class CcOrgaMemberCard extends withResizeObserver(LitElement) {
+
+  static get properties () {
+    return {
+      authorisations: { type: Object },
+      member: { type: Object },
+      _size: { type: String, state: true },
+      _newRole: { type: String, state: true },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {Authorisations} Sets the authorisations that control the display of the edit / delete buttons. */
+    this.authorisations = {
+      edit: false,
+      delete: false,
+    };
+
+    /** @type {OrgaMemberCardState} Sets the state and data of the member. */
+    this.member = {
+      state: 'loaded',
+      id: '',
+      email: '',
+      role: '',
+      isMfaEnabled: false,
+      isCurrentUser: false,
+    };
+
+    /** @protected */
+    this.breakpoints = {
+      // used to switch to buttons wrapping below when component width < 740 and vertical card layout when < 580, buttons and badges below each other when < 350.
+      width: [BREAKPOINT_TINY, BREAKPOINT_SMALL, BREAKPOINT_MEDIUM],
+    };
+
+    /** @type {Ref<CcInputText>} */
+    this._deleteButtonRef = createRef();
+
+    this._newRole = null;
+
+    /** @type {Ref<CcSelect>} */
+    this._roleRef = createRef();
+
+    /** @type {null} Set by `withResizeObserver` mixin. See the `onResize` method for more info. */
+    this._size = '';
+  }
+
+  /*
+  * Focus the delete button within the card.
+  * To be used when a card is removed, and you want to focus another card (LostFocusController used in `cc-orga-member-list`).
+  */
+  focusDeleteBtn () {
+    this._deleteButtonRef.value.focus();
+  }
+
+  /* Used by the `withResizeObserver` mixin. */
+  onResize ({ width }) {
+    this._size = width;
+  }
+
+  /* The accessible name provides more info than the visible text. It mentions the member being edited / to be edited. */
+  _getFirstBtnAccessibleName () {
```

Comment:
When you first read the component code `_*firstBtn*` and `_*secondBtn*` functions might seem unclear as you don't know at first that it is related to the edit button and the leave one. Do we want to change it to something more clearer for the future? :thinking:

Reply:
These buttons used to have classes "first-btn" and "second-btn" but I didn't need them anymore and I removed the classes.
I agree with you that it's not very clear but I can't change it to "edit" for instance because this button can be used to edit and cancel.
The second button is even worse because it might be used for delete, leave or validate.

This problem will disappear if we use 4 buttons within the DOM to solve the icon issue for slow network. I'll wait until we make a decision on this subject and if we keep the current system with 2 buttons, I'll try to find something clearer for the names :+1: